File: curl.js
const data = JSON.stringify({
  "lang": "fr",
  "title": "banane",
  "nPredLevels": 0,
  "nSuccLevels": 1,
  "reverseData": false,
  "keyword": "banane"
});

const xhr = new XMLHttpRequest();
xhr.withCredentials = true;

xhr.addEventListener("readystatechange", function () {
  if (this.readyState === this.DONE) {
    console.log(this.responseText);
  }
});

xhr.open("POST", "https://www.machinools.com/api/entity/neigh_graph");
xhr.setRequestHeader("Content-Type", "application/json");

xhr.send(data);

File: curl.py
import http.client

conn = http.client.HTTPSConnection("www.machinools.com")

payload = "{\n  \"lang\": \"fr\",\n  \"title\": \"banane\",\n  \"nPredLevels\": 0,\n  \"nSuccLevels\": 1,\n  \"reverseData\": false,\n  \"keyword\": \"banane\"\n}"

headers = { 'Content-Type': "application/json" }

conn.request("POST", "/api/entity/neigh_graph", payload, headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))
